packages/react-component-library/src/components/CheckboxRadioBase/CheckboxRadioBase.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useRef, useState, useEffect } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+import mergeRefs from 'react-merge-refs'
+
+import { CHECKBOX_RADIO_VARIANT } from './types'
+import { CheckboxRadioBaseProps } from './CheckboxRadioBaseProps'
+import { StyledWrapper } from './partials/StyledWrapper'
+import { StyledInnerWrapper } from './partials/StyledInnerWrapper'
+import { StyledDescription } from './partials/StyledDescription'
+import { StyledLabel } from './partials/StyledLabel'
+import { StyledInput } from './partials/StyledInput'
+
+export const CheckboxRadioBase = React.forwardRef<
+  HTMLInputElement,
+  CheckboxRadioBaseProps
+>(
+  (
+    {
+      className = '',
+      id = uuidv4(),
+      defaultChecked,
+      description,
+      isDisabled = false,
+      isInvalid,
+      label,
+      name,
+      onChange,
+      onBlur,
+      value,
+      variant = CHECKBOX_RADIO_VARIANT.DEFAULT,
+      type,
+      partials: { root: rootPartial, checkmark: checkmarkPartial },
+      ...rest
+    },
+    ref
+  ) => {
+    const localRef = useRef<HTMLInputElement>(null)
+    const [isChecked, setIsChecked] = useState(defaultChecked)
+
+    // Handle implicit deselection of radio in collection
+    useEffect(() => {
+      const handleSelected = (e: Event) => {
+        const {
+          detail: { ref: emittedRef },
+        } = e as CustomEvent
+
+        const isDifferent = emittedRef !== localRef
+        const isInCollection = emittedRef.current.attributes.name.value === name
+        const isRadio = type === 'radio'
+
+        if (isDifferent && isInCollection && isRadio) {
+          setIsChecked(false)
+        }
+      }
+
+      document.addEventListener('_radio:selected', handleSelected)
```

Comment:
Just a note for reference really, I think `document.addEventListener('change', handleSelected, {capture: true})` could've been made to work too with some checks on the target (but the custom event works fine so let's stick with that).